sdk/storage/azure-storage-blob/azure/storage/blob/_shared/avro/__init__.py

```diff
@@ -0,0 +1,5 @@
+# -------------------------------------------------------------------------
```

Comment:
Please note that the contents of the `_shared` directory must be copied across all storage SDKs (queues/fileshare/datalake/etc). This includes the `_shared` directory in the test suite. If you feel this should not be in the other SDKs (i.e. that avro will not be a part of queue messages, or datalake etc), then please move this directory up a layer :)

Reply:
@annatisch this code is going to be used in `azure-storage-blob` and `azure-storage-blob-changefeed` (that one is work in progress) modules. So shared seems to be better fit. If so do we need to copy avro to all storage modules or only two that take dependency on it?

For the context. It's imperative that we hide avro parser and not expose it. So having it packaged separately and adding as dependency isn't an option unfortunately.

Reply:
You can keep it in shared - but it will be distributed in every storage SDK (queues, datalake, etc).
At some point we will add a CI step to ensure the `_shared` module is kept consistent between all the SDKs.

If you do not wish this to be included in all the storage SDKs, then you could move it up a level to `azure.storage.blob._avro`

Reply:
Hi Anna! @annatisch 
We created a package azure-storage-blob-changefeed package which depends on azure-storage-blob. Previously I planned to add _shared folder to azure-storage-blob-changefeed to use avro(that is what I told kamil), but now we plan to directly import _shared.avro from azure-storage-blob so that we don't need to put _shared folder in blob-changefeed pkg.

In the future we will have changefeed feature in fileshare and datalake, so probably we can also import the _shared folder from azure-storage-file-share and azure-storage-file-datalake so we don't need to add _shared folder in azure-storage-fileshare-changefeed and azure-storage-datalake-changefeed(these changefeed packages haven't been created yet)

So I guess probably it makes more sense to put leave the avro in _shared folder? What do you think ^_^?

Reply:
So it stays in `_shared`.

Reply:
Thanks @xiafu-msft! Yes if the plan is to support this in files and datalake as well, then shared is a good place for it. Could you please duplicate it across the other packages?

Reply:
Hi Anna! @annatisch 
Thanks for your feedback! We still need to tweak the DafaFileReader a bit to fit ChangeFeed feature. After that we will duplicate it!